### PR TITLE
sql: implement migrate.StmtScanner in drivers

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: sql/migrate/lex.go [hunk 1]
```diff
@@ -54,9 +54,40 @@ func Stmts(input string) ([]*Stmt, error) {
 	}).Scan(input)
 }
 
+// FileStmtDecls scans atlas-format file statements using
+// the Driver implementation, if implemented.
+func FileStmtDecls(drv Driver, f File) ([]*Stmt, error) {
```
Comment (masseelch) [inline sql/migrate/lex.go:59]: Not sure I like the naming. I see `StmtDecls` is taken, but why `File`? We can extend the `StmtDecls` with a spread and use options like `Driver()` maybe?

File: sql/mysql/mysqlcheck/mysqlcheck.go [hunk 2]
```diff
@@ -142,12 +142,16 @@ func addNotNull(p *datadepend.ColumnPass) (diags []sqlcheck.Diagnostic, err erro
...
 	})
 	if !ok {
 		return false
 	}
-	filled, _ := p.ColumnFilledAfter(pass.File, pass.Table, pass.Column, pass.Change.Stmt.Pos, matchValue)
+	stmts, err := migrate.FileStmtDecls(pass.Dev, pass.File)
+	if err != nil {
+		return false
+	}
+	filled, _ := p.ColumnFilledAfter(stmts, pass.Table, pass.Column, pass.Change.Stmt.Pos, matchValue)
```
Comment (masseelch) [inline sql/mysql/mysqlcheck/mysqlcheck.go:154]: Ignore error?

**Review by rotemtam — APPROVED** [review]

File: sql/migrate/lex.go (hunk 1 above)
Comment (a8m) [reply sql/migrate/lex.go:59]: There's a method under `File` named `StmtDecls`. This is the same as calling File.StmtDecls, but accepts Driver as the first argument. Feels simpler and clear enough to me. Especially if I want to have both Stmts and StmtDecls.

File: sql/mysql/mysqlcheck/mysqlcheck.go (hunk 2 above)
Comment (a8m) [reply sql/mysql/mysqlcheck/mysqlcheck.go:154]: Yes, parsing can fail and we ignore this "enrichment" in that case (that logic was not changed).

**Pull request merged**